db/crud.go

```diff
@@ -869,6 +871,42 @@ func (db *DatabaseCollectionWithUser) OnDemandImportForWrite(ctx context.Context
 	return nil
 }
 
+// updateMutateInSpec returns the mutate in spec needed for the document update based off the outcome in updateHLV
+func updateMutateInSpec(hlv HybridLogicalVector) []sgbucket.MacroExpansionSpec {
+	var outputSpec []sgbucket.MacroExpansionSpec
+	if hlv.Version == math.MaxUint64 {
+		spec := sgbucket.NewMacroExpansionSpec(xattrCurrentVersionPath(base.SyncXattrName), sgbucket.MacroCas)
+		outputSpec = append(outputSpec, spec)
+	}
+	if hlv.CurrentVersionCAS == math.MaxUint64 {
+		spec := sgbucket.NewMacroExpansionSpec(xattrCurrentVersionCASPath(base.SyncXattrName), sgbucket.MacroCas)
+		outputSpec = append(outputSpec, spec)
+	}
+	return outputSpec
+}
+
+// updateHLV updates the HLV in the sync data appropriately based on what type of document update event we are encountering
+func (db *DatabaseCollectionWithUser) updateHLV(d *Document, docUpdateEvent DocUpdateEvent) (*Document, error) {
+
+	switch docUpdateEvent.eventType {
+	case BlipWriteEvent:
+		// preserve any other logic on the HLV that has been done by the client, only update to cvCAS will be needed
+		d.HLV.CurrentVersionCAS = hlvExpandMacroCASValue
+	case ImportEvent:
+		// VV should remain unchanged
```

Comment:
There are cases where import needs to update the VV (if the source of the mutation was a non-XDCR SDK write), but it's out of scope of this ticket.  Filed CBG-3503 for this work, can you mention that ticket in the comment?